cpp/arcticdb/python/python_handlers.cpp

```diff
@@ -53,33 +53,61 @@ namespace arcticdb {
         return dest + count;
     }
 
+	constexpr inline bool is_nullable_type(TypeDescriptor td) {
+		return td.dimension() > Dimension::Dim0 || is_empty_type(td.data_type()) || is_py_bool_type(td.data_type());
+	}
+
     void EmptyHandler::handle_type(
         const uint8_t*& input,
         uint8_t* dest,
         const VariantField& variant_field,
-        const entity::TypeDescriptor&,
+        const entity::TypeDescriptor& source_type_descriptor,
+        const entity::TypeDescriptor& destination_type_descriptor,
         size_t dest_bytes,
         std::shared_ptr<BufferHolder>,
         EncodingVersion
     ) {
         ARCTICDB_SAMPLE(HandleEmpty, 0)
         util::check(dest != nullptr, "Got null destination pointer");
-        const size_t num_rows = dest_bytes / get_type_size(DataType::EMPTYVAL);
-        static_assert(get_type_size(DataType::EMPTYVAL) == sizeof(PyObject *));
-        auto ptr_dest = reinterpret_cast<const PyObject **>(dest);
-        fill_with_none(ptr_dest, num_rows);
+        const size_t num_rows = dest_bytes / destination_type_descriptor.get_type_byte_size();
+        static_assert(get_type_size(DataType::EMPTYVAL) == sizeof(PyObject*));
+
+        // TODO: arcticdb::util::default_initialize does mostly the same thing
```

Comment:
You could check if it's a nullable object type and do the Python-specific stuff, otherwise forward to default_initialize() to avoid the repetition.